lib/LaTeXML/Engine/TeX_Kern.pool.ltxml

```diff
@@ -27,16 +27,40 @@ use LaTeXML::Package;
 # \lastkern         iq is 0.0 pt or the last kern on the current list.
 
 # \kern is heavily used by xy.
-# Completely HACK version for the moment
 # Note that \kern should add vertical spacing in vertical modes!
 DefConstructor('\kern Dimension', sub {
     my ($document, $length, %props) = @_;
     my $parent = $document->getNode;
     if ($document->getNodeQName($parent) eq 'svg:g') {
       if (my $x = $length->pxValue) {
-        # HACK HACK HACK
         my $transform = $parent->getAttribute('transform');
         $parent->setAttribute(transform => ($transform ? $transform . ' ' : '') . "translate($x,0)");
+        if (my @nodes = $parent->childNodes) {
+          # nodes have already been deposited, wrap those in svg:g to avoid shifting them
+          # the code below is essentially Document::wrapNodes, but we add the transform attribute
+          # *before* running afterClose, or collapseSVGGroup will immediately remove the wrapper
+
+          # Check if any of @nodes, or any of it's children, are the current node, and thus still "open"
+          my $leave_open = 0;
+          foreach my $n (@nodes) {
+            if ($document->isOpen($n)) {
+              $leave_open = 1;
+              last; } }
+
+          my $model = $document->getModel;
+          my ($ns, $tag) = $model->decodeQName('svg:g');
+          my $new = $document->openElement_internal($parent, $ns, $tag);
+          $new->setAttribute(transform => 'translate(' . -$x . ',0)');
+          $document->afterOpen($new);
+          $parent->replaceChild($new, $nodes[0]);
+
+          if (my $font = $document->getNodeFont($parent)) {
+            $document->setNodeFont($new, $font); }
+          if (my $box = $document->getNodeBox($parent)) {
+            $document->setNodeBox($new, $box); }
+          foreach my $node (@nodes) {
+            $new->appendChild($node); }
+          $document->afterClose($new) unless $leave_open; }
```

Comment:
So, if I understand the problem, if there are already children in the current `svg:g`, you don't want to modify the transform of *that* `g`, but what you really want to do is simply open a *new* one; the following children will go there and be translated, right?  You just need `_autoclose=1`, so that that new group will be closed appropriately.

However, if you *had* tried that, it would have failed, since there's some mixup with `svg:g` being opened and closed wrongly. I think I've sorted that out and can make an alternative PR, or try to explain where some other changes are needed (`TeX_Box` and `pgfsys-latexml`).

Reply:
> if you _had_ tried that, it would have failed

Yep, that would have been my preferred approach, but nodes would just disappear. Might have gain an extra gray hair in the process.

> I think I've sorted that out and can make an alternative PR

By all means, if you know how to tweak `svg:g` so that it closes at the right time, please do!